snowflake/gen.go

```diff
@@ -0,0 +1,131 @@
+// Package snowflake is copied directly from influxdb/pkg/snowflake to generate
+// a universally unique uint64.
+//
+// pkg/snowflake cannot be imported because there is no way to import
+// github.com/influxdata/influxdb/influxql@1.1.5 for this package nor
+// github.com/influxdata/influxql@3684fd4f35857877a7a9659e1dfea8bce026d08c for
+// other chronograf specific logic.
+package snowflake
+
+import (
+	"fmt"
+	"sync/atomic"
+	"time"
+)
+
+const (
+	epoch        = 1491696000000
+	serverBits   = 10
+	sequenceBits = 12
+	timeBits     = 42
+	serverShift  = sequenceBits
+	timeShift    = sequenceBits + serverBits
+	serverMax    = ^(-1 << serverBits)
+	sequenceMask = ^(-1 << sequenceBits)
+	timeMask     = ^(-1 << timeBits)
+)
+
+type Generator struct {
+	state   uint64
+	machine uint64
+}
+
+func New(machineID int) *Generator {
```

Comment:
Could you add some comments for all the exported functions? That will make goDoc very happy.

Reply:
Sure, I'll also make a PR to https://github.com/influxdata/influxdb where I pulled these from.